src/Client.php

```diff
@@ -37,32 +37,26 @@ public function __construct(CakeClient $client = null, ResponseFactory $response
      */
     public function sendRequest(RequestInterface $request)
     {
-        $cakeRequest = new Request();
-        $cakeRequest->method($request->getMethod());
-        $cakeRequest->url((string) $request->getUri());
-        $cakeRequest->version($request->getProtocolVersion());
-        $cakeRequest->body($request->getBody()->getContents());
+        $cakeRequest = new Request(
```

Comment:
Yes unfortunately Cake's HTTP client's method argument is typehinted as the concrete class `Cake\Http\Client\Request` instead of PSR7's interface. So for the time being this is required.